src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -135,6 +132,18 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
       DefDef(Modifiers(Flags.JAVA), nme.CONSTRUCTOR, List(), List(vparams), TypeTree(), blankExpr)
     }
 
+    def withConstructor(stats: List[Tree]): List[Tree] = {
+      val pureStats = stats map {
+        case DocDef(_, defn) => defn
+        case defn => defn
+      }
+      if(treeInfo.firstConstructor(pureStats) == EmptyTree) {
```

Comment:
Thanks for the tip! I initially tried unwrapping in `scala.reflect.internal.TreeInfo` however `DocDefs` are only available in package `scala.tools.nsc.ast`. With so many trees and infos around, it makes it hard to see the forest for the trees! :smile: